java/adapter/avro/src/main/java/org/apache/arrow/AvroToArrowUtils.java

```diff
@@ -221,16 +226,106 @@ private static Consumer createConsumer(
+          Schema nullSchema = schema.getTypes().stream().filter(s -> s.getType() == Type.NULL).findFirst().get();
+          int nullIndex = schema.getTypes().indexOf(nullSchema);
+          Schema subSchema = schema.getTypes().stream().filter(s -> s.getType() != Type.NULL).findFirst().get();
+          Preconditions.checkNotNull(subSchema, "schema should not be null.");
+          final Consumer consumer = createSkippableConsumer(subSchema);
+          return new NullableTypeConsumer(consumer, nullIndex);
+          // real union type
+        } else {
+
+          List<Consumer> delegates = schema.getTypes().stream().map(s ->
+              createSkippableConsumer(s)).collect(Collectors.toList());
+          skipFunction =  decoder -> delegates.get(decoder.readInt()).consume(decoder);
+
+          break;
+        }
+      case ARRAY:
+        skipFunction = decoder -> decoder.skipArray();
+        break;
+      case MAP:
+        skipFunction = decoder -> decoder.skipMap();
+        break;
+      case RECORD:
+        List<Consumer> delegates = schema.getFields().stream().map(field ->
+            createSkippableConsumer(field.schema())).collect(Collectors.toList());
+
+        skipFunction = decoder -> {
+          for (Consumer consumer : delegates) {
+            consumer.consume(decoder);
+          }
+        };
+
+        break;
+      case ENUM:
+        skipFunction = decoder -> decoder.readEnum();
+        break;
+      case STRING:
+        skipFunction = decoder -> decoder.skipString();
+        break;
+      case FIXED:
+        skipFunction = decoder -> decoder.skipFixed(schema.getFixedSize());
+        break;
+      case INT:
+        skipFunction = decoder -> decoder.readInt();
+        break;
+      case BOOLEAN:
+        skipFunction = decoder -> decoder.skipFixed(1);
+        break;
+      case LONG:
+        skipFunction = decoder -> decoder.readLong();
+        break;
+      case FLOAT:
+        skipFunction = decoder -> decoder.readFloat();
+        break;
+      case DOUBLE:
+        skipFunction = decoder -> decoder.readDouble();
+        break;
+      case BYTES:
+        skipFunction = decoder -> decoder.skipBytes();
+        break;
+      case NULL:
+        skipFunction = decoder -> { };
+        break;
+      default:
+        // no-op, shouldn't get here
+        throw new RuntimeException("Invalid avro type: " + type.getName());
+    }
+
+    return new SkippableConsumer(skipFunction);
+  }
+
   static CompositeAvroConsumer createCompositeConsumer(
       Schema schema, AvroToArrowConfig config) {
 
     List<Consumer> consumers = new ArrayList<>();
+    final List<String> skipFieldNames = config.getSkipFieldNames();
 
     Schema.Type type = schema.getType();
     if (type == Type.RECORD) {
       for (Schema.Field field : schema.getFields()) {
-        Consumer consumer = createConsumer(field.schema(), field.name(), config);
-        consumers.add(consumer);
+        if (skipFieldNames.contains(field.name())) {
```

Comment:
what happens if a parent and child field have the same name?  Is this possible?

Reply:
Avro not allow fields have same names, so this case is impossible.

Reply:
ok, could you add a comment here given a reference, and perhaps a unit test that demonstrates this (I would expect parsing a schema with the same name to fail.

Reply:
sorry to correct, seems same names are supported, schema below is valid:
>{
 "namespace": "org.apache.arrow.avro",
 "type": "record",
 "name": "testSkipNested",
 "fields": [
  {
    "name" : "f1",
    "type" : {
        "type" : "record",
        "name" : "nestedInRecord",
        "fields": [
             {"name": "f1", "type": "int"}
        ]
    }
  }
  ]
}

Both parent record field and child int field have the same name "f1", in this case, the whole record will be skipped when we met the first "f1"

Reply:
Do you think this is reasonable or we should do something to distinguish them (I think it's a little hard and messy)?

Reply:
I we should do something to distinguish them.

Reply:
Ok, i should think about this. can we make this as a follow-up?